### DISCUSSION TITLE
Official reference?

### DISCUSSION THREAD

**Discussion by resuni** [alexdebril/feed-io#367] [Q&A]
Does this project have any official reference documentation? It looks like at one point it may have been at https://feed-io.net/documentation.html but the website is currently down. 

Challenges I'm currently facing that I'm hoping documentation will solve:
* I need to determine the feed type and version (e.g. RSS 2.0) in addition to parsing the normal feed attributes. `$feedIo->read($feed)->getFeed()` does not return this information. 
* Discover function doesn't pick up JSON Feeds. Using NPR.org as an example, only RSS feeds are detected even though both RSS and JSON feeds are referenced via `<meta>` tags:
```
$ vendor/bin/feedio discover https://www.npr.org/
Discovering feeds from https://www.npr.org/
found : https://feeds.npr.org/1002/rss.xml
found : https://feeds.npr.org/1001/rss.xml
found : https://feeds.npr.org/1039/rss.xml
found : https://feeds.npr.org/1032/rss.xml
found : https://feeds.npr.org/3/rss.xml
found : https://feeds.npr.org/2/rss.xml
found : https://feeds.npr.org/35/rss.xml
```

**Comment by alexdebril** [comment]
I shut down feed-io.net as it wasn't relevant to keep it online (no traffic + it's more convenient to have everything in one place). Where did you see a reference to the old website ? 

So actually the official reference is the README.md file.

FeedIo doesn't expose the feed type, it detects it to pick the correct parser but "forgets" it right away. If you want to get this information maybe you can take a look at the classes in `Standard` namespace to figure things out and submit a pull request.

Not detecting JSON feeds is a bug, thanks for sharing that. I'll create an issue and fix it.

Thanks a lot for your interest in feed-io and for sharing this feedback

**Reply by resuni** [reply]
Pretty sure I wrote my reply in the wrong place. I've never used this GitHub discussions feature before. I copy/pasted it here and deleted the answer I originally left it in.

> I shut down feed-io.net as it wasn't relevant to keep it online (no traffic + it's more convenient to have everything in one place). Where did you see a reference to the old website ?

I found a link to it in an old issue you replied to years ago: https://github.com/alexdebril/feed-io/issues/149#issuecomment-348949227

feed-io.net is also listed as the homepage on Packagist: https://packagist.org/packages/debril/feed-io

I think I've seen it other places too. I'll let you know if I find them again. 

> FeedIo doesn't expose the feed type, it detects it to pick the correct parser but "forgets" it right away. If you want to get this information maybe you can take a look at the classes in Standard namespace to figure things out and submit a pull request.

I've never developed a composer package like this before, but maybe now is a good time to learn. I'll take a look and see what I can do. 

 > Not detecting JSON feeds is a bug, thanks for sharing that. I'll create an issue and fix it.

One thing I noticed about npr.org is when they reference the JSON feeds, they specify `content-type=application/json`, rather than the more specific `content-type=application/feed+json`. It doesn't seem wise to assume a reference with content-type application/json is always a JSON feed, so I don't see a way to solve the problem without following the reference and parsing the JSON. Regardless, I don't see either content-type listed in VALID_TYPES, but I'm not sure this is the right place to be looking: https://github.com/alexdebril/feed-io/blob/main/src/FeedIo/Explorer.php#L13
